pyqldb/errors/__init__.py

```diff
@@ -54,78 +57,101 @@ def __init__(self, error):
+
+
 def is_occ_conflict_exception(e):
     """
     Is the exception an OccConflictException?
 
-    :type e: :py:class:`botocore.exceptions.ClientError`
-    :param e: The ClientError caught.
+    :type e: :py:class:`builtins.Exception`
+    :param e: The Exception caught.
 
     :rtype: bool
     :return: True if the exception is an OccConflictException. False otherwise.
     """
-    is_occ = e.response['Error']['Code'] == 'OccConflictException'
-    return is_occ
+    if isinstance(e, ClientError):
+        is_occ = e.response['Error']['Code'] == 'OccConflictException'
+        return is_occ
+    return False
 
 
 def is_bad_request_exception(e):
     """
     Is the exception a BadRequestException?
 
-    :type e: :py:class:`botocore.exceptions.ClientError`
-    :param e: The ClientError caught.
+    :type e: :py:class:`builtins.Exception`
+    :param e: The Exception caught.
 
     :rtype: bool
     :return: True if the exception is an BadRequestException. False otherwise.
     """
-    is_bad_request = e.response['Error']['Code'] == "BadRequestException"
-    return is_bad_request
+    if isinstance(e, ClientError):
+        is_bad_request = e.response['Error']['Code'] == "BadRequestException"
+        return is_bad_request
+    return False
 
 
 def is_invalid_session_exception(e):
     """
     Is the exception an InvalidSessionException?
 
-    :type e: :py:class:`botocore.exceptions.ClientError`
-    :param e: The ClientError caught.
+    :type e: :py:class:`builtins.Exception`
+    :param e: The Exception caught.
 
     :rtype: bool
     :return: True if the exception is an InvalidSessionException. False otherwise.
     """
-    is_invalid_session = e.response['Error']['Code'] == 'InvalidSessionException'
-    return is_invalid_session
+    if isinstance(e, ClientError):
+        is_invalid_session = e.response['Error']['Code'] == 'InvalidSessionException'
+        return is_invalid_session
+    return False
 
 
 def is_transaction_expired_exception(e):
     """
     Does this exception denote that a transaction has expired?
 
-    :type e: :py:class:`botocore.exceptions.ClientError`
-    :param e: The ClientError caught.
+    :type e: :py:class:`builtins.Exception`
+    :param e: The Exception caught.
 
     :rtype: bool
     :return: True if the exception denote that a transaction has expired. False otherwise.
     """
-    is_invalid_session = e.response['Error']['Code'] == 'InvalidSessionException'
-
-    if "Message" in e.response["Error"]:
-        return is_invalid_session and re.search("Transaction .* has expired", e.response["Error"]["Message"])
+    # TODO CHECK IF THIS IS ClientError
+    if isinstance(e, ClientError):
+        is_invalid_session = e.response['Error']['Code'] == 'InvalidSessionException'
 
+        if "Message" in e.response["Error"]:
+            return is_invalid_session and re.search("Transaction .* has expired", e.response["Error"]["Message"])
     return False
 
 
 def is_retriable_exception(e):
```

Comment:
Is this a publicly available function, and hence we cannot change the wording?

Reply:
Thats correct